### PR TITLE
Add resource owner password credentials grant support

### PR DESCRIPTION
**Pull request by charlesgreen**
Hello @yaegashi, 

Great project. It's very helpful. 

I added support for Resource Owner Password Credentials in a project I'm using.
https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth-ropc

Please let me know if you'd like me to make any changes. 
Again, great work!  よろしくお願いします。

### CODE REVIEW

File: msauth/resource_owner_password_grant.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+		ClientSecret: clientSecret,
+		Endpoint:     endpoint,
+		Scopes:       scopes,
+	}
+	var err error
+	t, err := config.PasswordCredentialsToken(ctx, username, password)
+	ts := config.TokenSource(ctx, t)
+	if err != nil {
+		return nil, err
+	}
```
Comment (yaegashi) [inline msauth/resource_owner_password_grant.go]:
```suggestion
	t, err := config.PasswordCredentialsToken(ctx, username, password)
	if err != nil {
		return nil, err
	}
	ts := config.TokenSource(ctx, t)
```

- `var err error` is unneeded.
- It should return immediately if it gets an error.  `t` might cause nil pointer dereference.

**Review by yaegashi — CHANGES_REQUESTED** [review]
Hi, thanks for the patch!

Could you add the reference and a code snippet like other grant flows in README.md?

<pre>
### Resource owner password credentials grant

- [OAuth 2.0 resource owner passowrd credentials grant flow]

```go
// code snippet
```

[OAuth 2.0 resource owner passowrd credentials grant flow]: https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth-ropc
</pre>

File: msauth/resource_owner_password_grant.go (hunk 1 above)
Comment (charlesgreen) [reply msauth/resource_owner_password_grant.go]: Thanks for the feedback. I've made the changes and updated the README.

Comment (yaegashi) [reply msauth/resource_owner_password_grant.go]: Looks great.  Thank you!

**Review by yaegashi — APPROVED** [review]

**Comment by yaegashi** [comment]
Thank you for your contribution!

**Pull request merged**